builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,206 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
 
-    function App() {
-      const { contract } = useContract(contractAddress);
-      // Read data from your smart contract using the function or variables name
-      const { data, isLoading, error } = useContractRead(contract, 'INSERT_NAME');
-    }
-    ```
+```typescript title="App.tsx"
+import { readContract } from "thirdweb";
 
-- Use the [`useContractWrite` hook](https://portal.thirdweb.com/react/react.usecontractwrite){target=\_blank} to call any write functions on your contract by passing in the name of the function you want to use:
-
-    ```javascript
-    import {
-      useContractWrite,
-      useContract,
-      Web3Button,
-    } from '@thirdweb-dev/react';
-
-    // Your smart contract address
-    const contractAddress = 'INSERT_CONTRACT_ADDRESS';
-
-    function App() {
-      const { contract } = useContract(contractAddress);
-      const { mutateAsync, isLoading, error } = useContractWrite(
-        contract,
-        'INSERT_NAME'
-      );
-
-      return (
-        <Web3Button
-          contractAddress={contractAddress}
-          // Calls the 'INSERT_NAME' function on your smart contract
-          // with 'INSERT_ARGUMENT' as the first argument
-          action={() => mutateAsync({ args: ['INSERT_ARGUMENT'] })}
-        >
-          Send Transaction
-        </Web3Button>
-      );
-    }
-    ```
+const number = await readContract({
+      contract: contract,
+      method: "number",
+      params: [],
+    });
+```
 
 ### Connect Wallet {: #connect-wallet }
 
-There are a couple of ways that you can create a custom [connect wallet](https://portal.thirdweb.com/react/connecting-wallets){target=\_blank} experience. You can use the [`ConnectWallet` component](https://portal.thirdweb.com/react/connecting-wallets#using-the-connect-wallet-button){target=\_blank} or, for a more customizable approach, you can use the [`useConnect` hook](https://portal.thirdweb.com/react/connecting-wallets#using-hooks){target=\_blank}.
-
-The following example will show you how to use the `ConnectWallet` component. To go this route, you will need to specify the supported wallets and pass them to your provider.
-
-```javascript
-import {
-  ThirdwebProvider,
-  metamaskWallet,
-  coinbaseWallet,
-  walletConnectV1,
-  walletConnect,
-  safeWallet,
-  paperWallet,
-} from '@thirdweb-dev/react';
-
-function MyApp() {
+Next, let's customize the [Connect Button](https://portal.thirdweb.com/typescript/v5/react/components/ConnectButton){target=\_blank} to tailor it our desired wallets. You can add or remove wallets from the wallets array to change the options available to users. ThirdWeb also offers a [ConnectButton Playground](https://thirdweb.com/dashboard/connect/playground) to customize and view changes in real-time given the high degree of flexibility offered by the button. 
```

Comment:
let's

Reply:
If I am reading this without context and without looking at the code snippet below, I don't know what the wallets array is. So I'd just introduce it a bit by just saying something like, "The `ConnectButton` accepts a `wallets` parameter, which allows you to pass in an array containing the supported wallets."